### PR TITLE
Matched most of the os folder

### PR DESCRIPTION
**Pull request by Mr-Wiseguy**

### CODE REVIEW

File: src/os/getnextfaultthread.c [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    register OSThread *fault;
+
+    if (lastFault == NULL) {
+        fault = __osActiveQueue;
+    } else {
+        fault = lastFault;
+    }
+
+    while (fault->priority != -1) {
+        if ((fault->flags & 0x2) != 0 && fault != lastFault) {
```
Comment (Thar0) [inline src/os/getnextfaultthread.c:15]:
```suggestion
        if ((fault->flags & OS_FLAG_FAULT) && fault != lastFault) {
```

File: src/os/initialize.c [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    // Wait until there are no RCP interrupts
+    if (__osGetCause() & CAUSE_IP5) {
+        while (TRUE) {
+            ;
+        }
+    }
+
+    IO_WRITE(AI_CONTROL_REG, 1);
+    IO_WRITE(AI_DACRATE_REG, 0x3FFF);
+    IO_WRITE(AI_BITRATE_REG, 0xF);
```
Comment (Thar0) [inline src/os/initialize.c:81]:
```suggestion
    IO_WRITE(AI_CONTROL_REG, AI_CONTROL_DMA_ON);
    IO_WRITE(AI_DACRATE_REG, AI_MAX_DAC_RATE - 1);
    IO_WRITE(AI_BITRATE_REG, AI_MAX_BIT_RATE - 1);
```

File: src/os/resetglobalintmask.c [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#include <PR/os_internal.h>
+
+void __osResetGlobalIntMask(OSHWIntr mask) {
+    register u32 saveMask = __osDisableInt();
+
+    // TODO magic number
+    __OSGlobalIntMask &= ~(mask & ~0x401);
```
Comment (Thar0) [inline src/os/resetglobalintmask.c:7]:
```suggestion
    __OSGlobalIntMask &= ~(mask & ~OS_IM_RCP);
```

File: src/os/thread.c [hunk 4]
```diff
@@ -0,0 +1,25 @@
+#include <PR/os_internal.h>
+#include "osint.h"
+
+struct __osThreadTail __osThreadTail = {0, -1};
```
Comment (Thar0) [inline src/os/thread.c:4]:
```suggestion
struct __osThreadTail __osThreadTail = {NULL, -1};
```

File: src/os/timerintr.c [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+	__osTimerList->msg = 0;
+}
+
+void __osTimerInterrupt(void) {
+	OSTimer *t;
+	u32 count;
+	u32 elapsed_cycles;
+
+	if (__osTimerList->next != __osTimerList) {
+		while (1) {
```
Comment (Thar0) [inline src/os/timerintr.c:29]:
```suggestion
		while (TRUE) {
```

**Review by Thar0 — CHANGES_REQUESTED** [review]
Nice. Couple of little things.

File: include/PR/R4300.h [hunk 6]
```diff
@@ -1,302 +1,343 @@
-#ifndef _R4300_H_
```
[the remark this answers is not in the document]
Comment (Mr-Wiseguy) [reply include/PR/R4300.h:1]: Ah okay, I'll revert it then. I thought it was just an outdated version of the header like some of the other headers were.

[the remark this answers is not in the document]
Comment (Thar0) [reply include/PR/R4300.h:1]: Oh I didn't notice but the same is true for rcp.h and ultratypes.h (so all three headers in current master)

[the remark this answers is not in the document]
Comment (Mr-Wiseguy) [reply include/PR/R4300.h]: I'll do those as well then

**Review by Thar0 — APPROVED** [review]
LGTM

**Pull request merged**